modules/oe_content_persistent/src/ContentUuidResolver.php

```diff
@@ -0,0 +1,224 @@
+   * The alias manager that caches alias lookups based on the request.
+   *
+   * @var \Drupal\Core\Path\AliasManagerInterface
+   */
+  protected $aliasManager;
+
+  /**
+   * The cache backend.
+   *
+   * @var \Drupal\Core\Cache\CacheBackendInterface
+   */
+  protected $cache;
+
+  /**
+   * List of supported storages.
+   *
+   * @var array
+   */
+  protected $supportedStorages;
+
+  /**
+   * Constructs a ContentUuidResolver.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager.
+   * @param \Drupal\Core\Language\LanguageManagerInterface $language_manager
+   *   The language manager.
+   * @param \Drupal\Core\Path\AliasManagerInterface $alias_manager
+   *   The alias manager.
+   * @param \Drupal\Core\Cache\CacheBackendInterface $cache
+   *   The cache backend.
+   * @param array $supported_storages
+   *   List of supported storages.
+   */
+  public function __construct(EntityTypeManagerInterface $entity_type_manager, LanguageManagerInterface $language_manager, AliasManagerInterface $alias_manager, CacheBackendInterface $cache, array $supported_storages = []) {
+    $this->entityTypeManager = $entity_type_manager;
+    $this->languageManager = $language_manager;
+    $this->aliasManager = $alias_manager;
+    $this->cache = $cache;
+    $this->supportedStorages = $supported_storages;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setCacheKey($key) {
+    $this->cacheKey = $this->getCachePrefixKey() . $key;
+  }
+
+  /**
+   * Getting current cache key.
+   */
+  public function getCacheKey() {
+    return $this->cacheKey;
+  }
+
+  /**
+   * Clear static cache.
+   */
+  public function resetStaticCache(): void {
+    $this->uuidCacheTags = [];
+    $this->lookupMap = [];
+    $this->cacheNeedsWriting = FALSE;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function writeCache() {
+    // We assume that better to write cache only on request finishing
+    // on Controller page.
+    if ($this->cacheNeedsWriting === TRUE && !empty($this->cacheKey) && !empty($this->lookupMap)) {
+      $twenty_four_hours = 60 * 60 * 24;
+
+      if ($cache_tags = reset($this->uuidCacheTags)) {
+        $this->cache->set($this->cacheKey, reset($this->lookupMap), $this->getRequestTime() + $twenty_four_hours, $cache_tags);
+      }
+      else {
+        $this->cache->set($this->cacheKey, reset($this->lookupMap), $this->getRequestTime() + $twenty_four_hours);
+      }
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getAliasByUuid(string $uuid, string $langcode = NULL): ?string {
+    $langcode = $langcode ?: $this->languageManager->getCurrentLanguage(LanguageInterface::TYPE_URL)->getId();
+
+    // Here we are trying to use static cache.
+    if (!empty($this->lookupMap[$uuid])) {
+      return $this->lookupMap[$uuid];
+    }
+
+    // Using cache key initialized in controller
+    // (if we use service inside controller).
+    if ($this->cacheKey) {
```

Comment:
Perhaps move all persistent cache logic into a separate method

Reply:
Anyway for now we have decided to remove caching.